osc-server/src/main/java/org/osc/core/broker/service/validator/DeleteDistributedApplianceRequestValidator.java

```diff
@@ -53,13 +59,48 @@ public DistributedAppliance validateAndLoad(BaseDeleteRequest request) throws Ex
                             da.getId()));
         }
 
+        validateDAReferenceToSfcAndSecurityGroup(em, da);
+
         if (!da.getMarkedForDeletion() && request.isForceDelete()) {
             throw new VmidcBrokerValidationException(
                     "Distributed Appilance with ID "
                             + request.getId()
                             + " is not marked for deletion and force delete operation is applicable only for entries marked for deletion.");
         }
-
         return da;
     }
+
+    private void validateDAReferenceToSfcAndSecurityGroup(EntityManager em, DistributedAppliance da) throws Exception {
+        for (VirtualSystem vs : da.getVirtualSystems()) {
+            try {
+                validateVSReferenceToSfcAndSecurityGroup(em, vs);
+            } catch (VmidcBrokerValidationException ex) {
+                throw new VmidcBrokerValidationException(ex.getMessage() + " with name " + da.getName());
```

Comment:
this will be a user facing message...we should format this better.

The message will show
`Cannot delete distributed appliance which is currently referencing Service Function Chain 'foosfc' and binded to a SecurityGroup(s) 'foosg' with name 'fooda'`

I would just avoid this catch block or format the message like below 

`Cannot delete distributed appliance 'fooda' which is currently referencing Service Function Chain 'foosfc' and binded to a SecurityGroup(s) 'foosg'.

Reply:
+1 for the suggested message.  The main purpose of the catch is to avoid passing the da as a param just to have it in the message, makes the signature of the method validateVSReferenceToSfcAndSecurityGroup harder to understand.